### PR TITLE
[NIFI-13061] replace actions with menu

### PR DESCRIPTION
**Pull request by scottyaslan**

### CODE REVIEW

**Comment by rfellows** [comment]
will review

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/users/ui/user-listing/user-table/user-table.component.html [hunk 1]
```diff
@@ -79,24 +79,37 @@
...
+                            ) {
+                                <button
+                                    mat-icon-button
+                                    type="button"
+                                    [matMenuTriggerFor]="actionMenu"
+                                    class="h-16 w-16 flex items-center justify-center icon global-menu">
+                                    <i class="fa fa-ellipsis-v"></i>
+                                </button>
                             }
+                            <mat-menu #actionMenu="matMenu" xPosition="before">
```
Comment (rfellows) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/users/ui/user-listing/user-table/user-table.component.html:93]: This menu doesn't close when the edit or user policies dialogs pop up in front:
Screenshot 2024-05-07 at 5 02 17 PM

It does hide if the delete confirmation dialog opens though.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/users/ui/user-listing/user-table/user-table.component.html [hunk 2]
```diff
@@ -79,24 +79,37 @@
...
+                                @if (canEditOrDelete(currentUser, item)) {
+                                    <button mat-menu-item (click)="deleteClicked(item)">
+                                        <i class="fa fa-trash primary-color mr-2"></i>
+                                        Remove
+                                    </button>
+                                }
+                                @if (hasAccessPolicies(item)) {
+                                    <button mat-menu-item (click)="viewAccessPoliciesClicked(item, $event)">
+                                        <i class="fa fa-key primary-color mr-2"></i>
+                                        View User Policies
```
Comment (rfellows) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/users/ui/user-listing/user-table/user-table.component.html:109]: We should be consistent in our menu options for policies. This is slightly different than the "Access Policies" option, but however we decide to phrase that one should inform this as well so they are similar

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.html [hunk 3]
```diff
@@ -116,62 +116,73 @@
...
+                                @if (canChangeVersion(item)) {
+                                    <button mat-menu-item (click)="changeVersionClicked(item)">
+                                        <i class="fa fa-exchange primary-color mr-2"></i>
+                                        Change Version
+                                    </button>
+                                }
+                                @if (canDelete(item)) {
+                                    <button mat-menu-item (click)="deleteClicked(item, $event)">
+                                        <i class="fa fa-trash primary-color mr-2"></i>
+                                        Delete
```
Comment (rfellows) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.html:160]: Clicking the delete or edit action does not close the menu:
Screenshot 2024-05-07 at 5 09 18 PM

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/reporting-tasks/reporting-task-table/reporting-task-table.component.html [hunk 4]
```diff
@@ -111,53 +111,64 @@
...
+                            mat-icon-button
+                            type="button"
+                            [matMenuTriggerFor]="actionMenu"
+                            class="h-16 w-16 flex items-center justify-center icon global-menu">
+                            <i class="fa fa-ellipsis-v"></i>
+                        </button>
+                        <mat-menu #actionMenu="matMenu" xPosition="before">
+                            <button mat-menu-item (click)="configureClicked(item, $event)">
+                                <i class="fa fa-cog primary-color mr-2"></i>
+                                {{ canEdit(item) ? 'Edit' : 'View Configuration' }}
```
Comment (rfellows) [inline nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/reporting-tasks/reporting-task-table/reporting-task-table.component.html:124]: Edit opens the dialog but does not close the menu:
Screenshot 2024-05-07 at 5 11 35 PM

**Review by rfellows — CHANGES_REQUESTED** [review]
I found a number of instances where selecting a menu option doesn't close the menu and it is still visible behind the dialog that it opens. I commented inline in a number of places, but there are more. I just stopped as it was getting redundant.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.html [hunk 5]
```diff
@@ -116,62 +116,73 @@
...
+                                        View State
+                                    </button>
+                                }
+                                @if (canManageAccessPolicies()) {
+                                    <button
+                                        mat-menu-item
+                                        (click)="$event.stopPropagation()"
+                                        [routerLink]="getPolicyLink(item)">
+                                        <i class="fa fa-key primary-color mr-2"></i>
+                                        Manage Access Policies
```
[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.html:175]: I will update all to "Manage Access Policies"

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/processor-status-listing/processor-status-table/processor-status-table.component.html [hunk 6]
```diff
@@ -248,25 +248,36 @@
...
+                                                mat-menu-item
+                                                [routerLink]="getProcessorLink(item)"
+                                                (click)="$event.stopPropagation()">
+                                                <i class="fa fa-long-arrow-right primary-color mr-2"></i>
+                                                Go to Processor in
+                                                {{ item?.processorStatusSnapshot?.processGroupNamePath }}
+                                            </button>
+                                            <button mat-menu-item (click)="viewStatusHistoryClicked($event, item)">
+                                                <i class="fa fa-area-chart primary-color mr-2"></i>
+                                                View Status History
```
[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/processor-status-listing/processor-status-table/processor-status-table.component.html:269]: The "View Status History" seems to be correct. For the clustered details I will update those menu options to "View Clustered Details" everywhere.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.html (hunk 3 above)
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.html:160]: This happens anywhere we `event.stopPropagation();`. I think these were in place to not cause selection to fire. I will remove them.

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/reporting-tasks/reporting-task-table/reporting-task-table.component.html (hunk 4 above)
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/reporting-tasks/reporting-task-table/reporting-task-table.component.html:124]: https://github.com/apache/nifi/pull/8766#discussion_r1594323472

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/users/ui/user-listing/user-table/user-table.component.html (hunk 1 above)
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/users/ui/user-listing/user-table/user-table.component.html:93]: https://github.com/apache/nifi/pull/8766#discussion_r1594323472

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/processor-status-listing/processor-status-table/processor-status-table.component.html [hunk 7]
```diff
@@ -248,25 +248,36 @@
...
-                                                item?.processorStatusSnapshot?.processGroupNamePath
-                                            }}"></div>
+                                        <button
+                                            mat-icon-button
+                                            type="button"
+                                            [matMenuTriggerFor]="actionMenu"
+                                            class="h-16 w-16 flex items-center justify-center icon global-menu">
+                                            <i class="fa fa-ellipsis-v"></i>
+                                        </button>
+                                        <mat-menu #actionMenu="matMenu" xPosition="before">
```
[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/summary/ui/processor-status-listing/processor-status-table/processor-status-table.component.html:258]: https://github.com/apache/nifi/pull/8766#discussion_r1594323472

File: nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/parameter-contexts/ui/parameter-context-listing/parameter-context-table/parameter-context-table.component.html [hunk 8]
```diff
@@ -58,32 +58,52 @@
...
+                                class="h-16 w-16 flex items-center justify-center icon global-menu">
+                                <i class="fa fa-ellipsis-v"></i>
+                            </button>
                         }
+                        <mat-menu #actionMenu="matMenu" xPosition="before">
+                            @if (canRead(item)) {
+                                <button mat-menu-item (click)="editClicked(item, $event)">
+                                    <i class="fa fa-cog primary-color mr-2"></i>
+                                    {{ canWrite(item) ? 'Edit' : 'View Configuration' }}
+                                </button>
```
[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/parameter-contexts/ui/parameter-context-listing/parameter-context-table/parameter-context-table.component.html]: https://github.com/apache/nifi/pull/8766#discussion_r1594323472

**Comment by scottyaslan** [comment]
> I found a number of instances where selecting a menu option doesn't close the menu and it is still visible behind the dialog that it opens. I commented inline in a number of places, but there are more. I just stopped as it was getting redundant.

Yes, I see that this happens anytime we `event.stopPropagation();`. I have removed them as they are no longer necessary.

**Review by rfellows — COMMENTED** [review]
Just one thing.

**Review by rfellows — APPROVED** [review]
Looks good. Thanks, @scottyaslan

**Pull request merged**